From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 3 Feb 2020 20:07:49 +0100
Subject: [PATCH] {BUG in 5.6-rc1} drivers/net/wireless/ath/ath11k: initialize
 rate_idx

clang reports on x86_64:

    drivers/net/wireless/ath/ath11k/debugfs_sta.c:185:7: error: variable
    'rate_idx' is used uninitialized whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
                    if (mcs > 9) {
                        ^~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:201:57: note: uninitialized use occurs here
            ath11k_accumulate_per_peer_tx_stats(arsta, peer_stats, rate_idx);
                                                                   ^~~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:185:3: note: remove
    the 'if' if its condition is always true
                    if (mcs > 9) {
                    ^~~~~~~~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:184:13: error: variable
    'rate_idx' is used uninitialized whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
            } else if (pkt_type == HAL_TX_RATE_STATS_PKT_TYPE_11AC) {
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:201:57: note: uninitialized use occurs here
            ath11k_accumulate_per_peer_tx_stats(arsta, peer_stats, rate_idx);
                                                                   ^~~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:184:9: note: remove
    the 'if' if its condition is always true
            } else if (pkt_type == HAL_TX_RATE_STATS_PKT_TYPE_11AC) {
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:175:7: error: variable
    'rate_idx' is used uninitialized whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
                    if (mcs > 7) {
                        ^~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:201:57: note: uninitialized use occurs here
            ath11k_accumulate_per_peer_tx_stats(arsta, peer_stats, rate_idx);
                                                                   ^~~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:175:3: note: remove
    the 'if' if its condition is always true
                    if (mcs > 7) {
                    ^~~~~~~~~~~~~
    drivers/net/wireless/ath/ath11k/debugfs_sta.c:139:13: note:
    initialize the variable 'rate_idx' to silence this warning
            u8 rate_idx;
                       ^
                        = '\0'
---
 drivers/net/wireless/ath/ath11k/debugfs_sta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath11k/debugfs_sta.c b/drivers/net/wireless/ath/ath11k/debugfs_sta.c
index 743760c9bcae..8238f9809b0e 100644
--- a/drivers/net/wireless/ath/ath11k/debugfs_sta.c
+++ b/drivers/net/wireless/ath/ath11k/debugfs_sta.c
@@ -136,7 +136,7 @@ void ath11k_update_per_peer_stats_from_txcompl(struct ath11k *ar,
 	struct ath11k_sta *arsta;
 	struct ieee80211_sta *sta;
 	u16 rate;
-	u8 rate_idx;
+	u8 rate_idx = 0; /* BUG */
 	int ret;
 	u8 mcs;
 
-- 
